src/components/DatePicker.vue

```diff
@@ -504,9 +511,27 @@
           return D + ['th', 'st', 'nd', 'rd'][D % 10 > 3 ? 0 : (D - D % 10 !== 10) * D % 10];
         }
       };
+      let checkinDate = this.formatDate(new Date(this.checkIn) , 'YYYYMM');
+      let startDate = this.formatDate(new Date(this.startDate), 'YYYYMM');
+      let endDate = this.formatDate(this.getNextMonth(new Date(this.startDate)),'YYYYMM');
+      if(this.checkIn && (parseInt(endDate) < parseInt(checkinDate) || parseInt(startDate) < parseInt(checkinDate)) ){
```

Comment:
instead of formatting the date to `YYYYMM` and using parseInt to compare two dates, we should probably use the `compareDay` method. It can be extracted to the `helpers` file to be shared between the `Day` and `Datepicker` components.

Reply:
variables removed,